generator/sbpg/targets/resources/c/test/legacy/sbp_cpp_test.cc.j2

```diff
@@ -28,6 +28,13 @@
 #include <libsbp/legacy/cpp/payload_handler.h>
 #include <libsbp/legacy/cpp/legacy_state.h>
 
+template<typename T, typename U = std::remove_reference_t<T>>
```

Comment:
Changes in this template required because fields in the wire encoded byte stream rarely have the correct alignment. Eg, a 2 byte integer has an alignment of `2` (on typical systems), but the starting address of the field in the byte stream might not be properly aligned.

x86 tends to cope with this properly which is why this hasn't caused problems so far, but it is a problem detectable by one of the sanitizers. This solution adds a little helper function to `memcpy` a few bytes from the bytestream in to a stack variable which is guaranteed to be properly aligned, then return to the caller.

This method reduces the amount of code changes required, although it does lead to a rather annoying use of `decltype` a few lines down. Since this is in legacy code and will be deleted soon it seems a reasonable solution.